tests/catch/unit/memory/hipHostGetFlags.cc

```diff
@@ -38,57 +64,171 @@ This testcase verifies hipHostGetFlags API basic scenario
-    HIP_CHECK(hipHostMalloc(reinterpret_cast<void**>(&C_h), SIZE,
-                            hipHostMallocMapped));
-
-    unsigned int flagA, flagB, flagC;
-    HIP_CHECK(hipHostGetDevicePointer(reinterpret_cast<void**>(&A_d), A_h, 0));
-    HIP_CHECK(hipHostGetDevicePointer(reinterpret_cast<void**>(&B_d), B_h, 0));
-    HIP_CHECK(hipHostGetDevicePointer(reinterpret_cast<void**>(&C_d), C_h, 0));
+    // Allocate using the generated flags combos
+    INFO("Flag passed when allocating: 0x" << std::hex << FlagComp << "\n");
+    HIP_CHECK(hipHostMalloc(reinterpret_cast<void**>(&A_h), SIZE, FlagComp));
+    unsigned int flagA{};
+
+    // get the flags from allocations and check if they are the same as the one set
     HIP_CHECK(hipHostGetFlags(&flagA, A_h));
-    HIP_CHECK(hipHostGetFlags(&flagB, B_h));
-    HIP_CHECK(hipHostGetFlags(&flagC, C_h));
 
-    HipTest::setDefaultData<TestType>(LEN, A_h, B_h, C_h);
+    checkFlags(FlagComp, flagA);
+    HIP_CHECK(hipHostFree(A_h));
+  }
+}
+
+// Test Allocation with flags and getting flags in another thread
+TEST_CASE("Unit_hipHostGetFlags_DifferentThreads") {
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-54");
+  return;
+#endif
+  constexpr auto SIZE{LEN * sizeof(int)};
+  int* A_h{nullptr};
+
+  const unsigned int FlagPart1 = GENERATE(from_range(FlagPart1Vec.begin(), FlagPart1Vec.end()));
+  const unsigned int FlagPart2 = GENERATE(from_range(FlagPart2Vec.begin(), FlagPart2Vec.end()));
 
-    dim3 dimGrid(LEN / 512, 1, 1);
-    dim3 dimBlock(512, 1, 1);
-    hipLaunchKernelGGL(HipTest::vectorADD, dimGrid, dimBlock,
-                       0, 0, static_cast<const TestType*>(A_d),
-                       static_cast<const TestType*>(B_d), C_d, LEN);
 
-    HIP_CHECK(hipMemcpy(C_h, C_d, SIZE, hipMemcpyDeviceToHost));
-    // Note this really HostToHost not
-    // DeviceToHost, since memory is mapped...
-    HipTest::checkVectorADD(A_h, B_h, C_h, LEN);
+  unsigned int FlagComp = FlagPart1 | FlagPart2;
+
+  hipDeviceProp_t prop;
+  int device{};
+  HIP_CHECK(hipGetDevice(&device));
+  HIP_CHECK(hipGetDeviceProperties(&prop, device));
+  if (prop.canMapHostMemory != 1) {
+    HipTest::HIP_SKIP_TEST("Device Property canMapHostMemory is not set");
+    return;
+  } else {
+    // Make sure we allocate before trying to get the flags
+    std::thread malloc_thread(
+        [&]() { HIP_CHECK_THREAD(hipHostMalloc(reinterpret_cast<void**>(&A_h), SIZE, FlagComp)); });
+    malloc_thread.join();
+    HIP_CHECK_THREAD_FINALIZE();
+    unsigned int flagA{};
+    HIP_CHECK(hipHostGetFlags(&flagA, A_h));
+
+    checkFlags(FlagComp, flagA);
 
-    REQUIRE(flagA == FlagA);
-    REQUIRE(flagB == FlagB);
-    REQUIRE(flagC == FlagC);
     HIP_CHECK(hipHostFree(A_h));
-    HIP_CHECK(hipHostFree(B_h));
-    HIP_CHECK(hipHostFree(C_h));
+  }
+}
+
+// Test behaviour of hipHostGetFlags with invalid args
+TEST_CASE("Unit_hipHostGetFlags_InvalidArgs") {
+#if HT_AMD
+  HipTest::HIP_SKIP_TEST("EXSWCPHIPT-54");
```

Comment:
Is it not good to pass the clear reason why it is skipped? With "EXSWCPHIPT-54", it is hard to decipher why is it getting skipped on AMD.